Shared/ContentView.swift

```diff
@@ -49,24 +39,46 @@ struct ContentView: View {
                 EmptyView()
             }
         }
+            
+        .fileImporter(isPresented: $presentFiles,
+                      allowedContentTypes: [.audio],
+                      allowsMultipleSelection: true,
+                      onCompletion: addSongHandler)
+    }
+    
+    func addSongHandler(result: Result<[URL], Error>?) {
+        guard let urls = try? result?.get() else { return print("Couldn't open files") }
+        
+        openSong (urls: urls)
         
     }
```

Comment:
Without context, I can only understand that this is reading in some audio files to open? Would love more context on what this thing is in general

Reply:
That's exactly what's going on yeah

Reply:
Sweet! 

So one thing to consider - what if only one of the files fails to open? Not sure if this is a use case for you